socketshark/subscription.py

```diff
@@ -78,6 +81,38 @@ def prepare_service_data(self):
         return await self.perform_service_request('on_unsubscribe',
                                                   raise_error=False)
 
+    def filter_message(self, data):
+        """
+        Returns whether to filter the given message.
+        """
+        # Check whether the message is filtered by comparing any defined
+        # filter_fields to auth_info.
+        filter_fields = self.service_config.get('filter_fields', [])
+        for field in filter_fields:
+            if field in data:
+                if self.session.auth_info.get(field) != data[field]:
+                    # Message filtered.
+                    print('filtered based on', field)
+                    return True
+
+        # Check whether the message is out-of-order.
+        if '_order' in data:
+            key = data.get('_order_key')
+            last_order = self.order_state.get(key)
+
+            try:
+                order = int(data['_order'])
+            except (TypeError, ValueError):
+                return True  # Filter invalid orders.
+
+            if last_order is not None and order <= last_order:
+                # Message filtered.
+                return True
+
+            self.order_state[key] = order
```

Comment:
It should get cleaned up when the subscription is cleaned up. How do you see a memory leak? The destructor (`Subscription.__del__`) gets called in my tests when unsubscribing, and I'd assume the order dict would get cleaned up with it.